target_chains/solana/programs/oracle-instance/src/lib.rs

```diff
@@ -0,0 +1,131 @@
+use {
+    anchor_lang::prelude::*,
+    pyth_solana_receiver::{
+        cpi::accounts::PostUpdate,
+        program::PythSolanaReceiver,
+        PostUpdateParams,
+    },
+    pyth_solana_receiver_sdk::{
+        config::{
+            Config,
+            DataSource,
+        },
+        price_update::{
+            PriceUpdateV1,
+            VerificationLevel,
+        },
+    },
+    pythnet_sdk::{
+        accumulators::merkle::MerkleRoot,
+        hashers::keccak256_160::Keccak160,
+        messages::{
+            FeedId,
+            Message,
+        },
+        wire::{
+            from_slice,
+            v1::{
+                MerklePriceUpdate,
+                WormholeMessage,
+                WormholePayload,
+            },
+        },
+    },
+    solana_program::{
+        keccak,
+        program_memory::sol_memcpy,
+        secp256k1_recover::secp256k1_recover,
+        system_instruction,
+    },
+    wormhole_core_bridge_solana::{
+        sdk::{
+            legacy::AccountVariant,
+            VaaAccount,
+        },
+        state::GuardianSet,
+    },
+    wormhole_raw_vaas::{
+        utils::quorum,
+        GuardianSetSig,
+        Vaa,
+    },
+};
+
+// pub mod error;
+pub mod sdk;
+
+declare_id!("F9SP6tBXw9Af7BYauo7Y2R5Es2mpv8FP5aNCXMihp6Za");
+
+#[error_code]
+pub enum OracleInstanceError {
+    #[msg("Updates must be monotonically increasing")]
+    UpdatesNotMonotonic,
+    #[msg("Price feed id mismatch")]
+    PriceFeedMessageMismatch
+}
+#[program]
+pub mod oracle_instance {
```

Comment:
i don't love this name, though I also can't think of a name that I really like. The best I have is "pyth-push-oracle", though that does risk being confused with the existing push oracle. Maybe "pyth-push-oracle-v2" (ugh)?

Reply:
pyth-push-oracle